library/network_connections.py

```diff
@@ -531,6 +531,16 @@ def ifcfg_create(
                 line = r["network"] + "/" + str(r["prefix"])
                 if r["gateway"]:
                     line += " via " + r["gateway"]
+                if connection["interface_name"]:
```

Comment:
I am not sure whether ansible always set `interface_name` or not, but this might lead to `KeyError`. Try `connection.get("interface_name")`

Reply:
this is not correct.

It's not about "whether ansible [...] set `interface_name`. The `connection` dictionary is the result of `ArgValidator.validate()` which validates and normalizes the dictionary. The major point of normalization is that the code later can rely on `connection` containing certain values and not need to do another round of error checking.

Also, the code above already accesses `connection["interface_name"]`